internal/replacements/replacements.go

```diff
@@ -257,6 +265,51 @@ func resourcesNeedsReplacement(desired []corev1.Container, current []corev1.Cont
 	return desiredCPURequests.Cmp(*currentCPURequests) == 1 || desiredMemoryRequests.Cmp(*currentMemoryRequests) == 1
 }
 
+// fileSecurityContextChanged checks for changes in the effective security context by checking that there are no changes
+// to the following SecurityContext (or PodSecurityContext) fields:
+// RunAsGroup, RunAsUser, FSGroup, or FSGroupChangePolicy
+// See https://github.com/FoundationDB/fdb-kubernetes-operator/issues/208 for motivation
+// only makes sense if both pods have containers with matching names
+func fileSecurityContextChanged(desired, current *corev1.Pod) bool {
+	// first check for FSGroup or FSGroupChangePolicy changes as that cannot be overridden at container level
+	// (if pod security context is identical, skip these checks)
+	if (desired.Spec.SecurityContext != nil || current.Spec.SecurityContext != nil) &&
+		!equality.Semantic.DeepEqualWithNilDifferentFromEmpty(desired.Spec.SecurityContext, current.Spec.SecurityContext) {
+		if desired.Spec.SecurityContext == nil { // check if changed non-nil -> nil
```

Comment:
```suggestion
		if desired.Spec.SecurityContext == nil && current.Spec.SecurityContext != nil { // check if changed non-nil -> nil
```

Otherwise the check below could cause a nil pointer exception?